### ISSUE TITLE
Importing json files is generating error

### ISSUE THREAD

**Issue by mdurini-swisscom** [Micke-K/IntuneManagement#267]
Hello, i exported some stuff  from a tenant and i'm trying to import them in another one. I'm getting these errors

Failed to invoke MS Graph with URL https://graph.microsoft.com/beta/deviceManagement/configurationPolicies (Request ID: 10af5dc4-c738-4edf-bbf7-07353ebe165b). Status code: BadRequest. Response message: The request is invalid. Exception: The remote server returned an error: (400) Bad Request.

Failed to invoke MS Graph with URL https://graph.microsoft.com/beta/deviceManagement/configurationPolicies (Request ID: 81688f33-b78d-4371-ac9b-eea45ae02c7f). Status code: BadRequest. Response message: A type named 'microsoft.graph.windows10GeneralConfiguration' could not be resolved by the model. When a model is available, each type name must resolve to a valid type. Exception: The remote server returned an error: (400) Bad Request.

Failed to invoke MS Graph with URL https://graph.microsoft.com/beta/deviceManagement/configurationPolicies (Request ID: d616ca41-5582-4b75-839a-1adcabfb1280). Status code: BadRequest. Response message: A type named 'microsoft.graph.windowsQualityUpdateProfile' could not be resolved by the model. When a model is available, each type name must resolve to a valid type. Exception: The remote server returned an error: (400) Bad Request.

this is also happening for powershell scripts. what am i doing wrong?

**Comment by mdurini-swisscom** [comment]
hello @Micke-K , first of all thank you for your amazing job. it seems that the problem is related to the policies
- Windows 10 and later updates
- Windows | Scripts and remediations (Powershell)

the funny thing is that they failed trying the export / import options, but it's working with bulk import / bulk export....do you need other info?

**Comment by Micke-K** [comment]
Thank you!

That sounds like you try to import them from a non-standard folder. The folder name specifies what API to use. The Bulk Export/Import takes care of that automatically. 

Can you do a manual import from a Bulk exported folder? 

Cheers!

**Comment by mdurini-swisscom** [comment]
So, when i did the export, i used these settings
![image](https://github.com/user-attachments/assets/13dd5b51-7af2-4bf1-a9c1-60a70c82b2ad)
so the folder structure was correct. 
If i use the import option and i select the company folder, i got these options
![image](https://github.com/user-attachments/assets/7c66cb6a-705d-49f4-b0ab-af9e7fdb1379)
if i select the "PowershellScripts" folder, i got these options
![image](https://github.com/user-attachments/assets/4a0be423-5348-4432-bdbb-e2fd44325c75)

**Comment by Micke-K** [comment]
That is weird. If Bulk Import works, then manual import should work from the same folder. The same import functions are used. Unless it was a temporary glitch in MS Graph. 

Does the log have any other issues?

Cheers!

**Comment by chamtl** [comment]
Hi,
I have the same issue
/Claus

**Comment by granwalla** [comment]
I'm having the same issue, too. I imported a few things to Compliance Policies and that worked. Importing a single item gives a very similar error (probably the same, I don't have it up right now).

PS: You're awesome and I love this tool.

**Comment by granwalla** [comment]
I was having this issue. Turns out I was trying to import something from Settings Catalog to Device Configuration. Maybe that will help someone else. :)

**Comment by Overlordkiwi** [comment]
I am having the same issue.  Import from MS Intune portal page is also failing.

**Comment by Micke-K** [comment]
Hello,

So it looks like Microsoft is doing some changes in the background. 

@Overlordkiwi what was the policy that got that error. 

I'll see if I can have a look on today or tomorrow. Down with a cold at moment but It try. 

Cheers!

**Comment by Micke-K** [comment]
I did a very quick test. Created a new PowerShell script and Windows Update policies.
Exported old ones I had + the new ones. I then Imported them again and all was successful. 

Can anyone upload a policy that is not working?

Cheers!

**Comment by Overlordkiwi** [comment]
________________________________

**Comment by Overlordkiwi** [comment]
[DoD Adobe Acrobat Pro DC Continuous STIG v2r1 Administrative Template.json](https://github.com/user-attachments/files/17082786/DoD.Adobe.Acrobat.Pro.DC.Continuous.STIG.v2r1.Administrative.Template.json)
[Google Chrome Update Policy Administrative Template.json](https://github.com/user-attachments/files/17082787/Google.Chrome.Update.Policy.Administrative.Template.json)

Thanks for taking a look. 

IS this the settings field that missing? under settingscount

PS C:\Users\Night> get-MgBetaDeviceManagementConfigurationPolicy -DeviceManagementConfigurationPolicyId 17117e74-14bb-4402-83f1-b3053045d91d | fl


Assignments          :
CreatedDateTime      : 7/5/2024 1:09:35 AM
CreationSource       :
Description          :
Id                   : 17117e74-14bb-4402-83f1-b3053045d91d
IsAssigned           :
LastModifiedDateTime : 7/5/2024 1:09:35 AM
Name                 : Steve's Test LAPS Policy
Platforms            : windows10
PriorityMetaData     : Microsoft.Graph.Beta.PowerShell.Models.MicrosoftGraphDeviceManagementPriorityMetaData
RoleScopeTagIds      : {0}
SettingCount         : 5
Settings             :
Technologies         : mdm
TemplateReference    : Microsoft.Graph.Beta.PowerShell.Models.MicrosoftGraphDeviceManagementConfigurationPolicyTemp
                       lateReference
AdditionalProperties : {[@odata.context,
                       https://graph.microsoft.com/beta/$metadata#deviceManagement/configurationPolicies/$entity]}

**Comment by Micke-K** [comment]
Hello,

Thank you for the files. I just had a look.

These are not exported with the tool. These json files only contains the settings and not the policy information. 

You would need a full export to be able to do the import.

Cheers!

**Comment by Overlordkiwi** [comment]
Yea, I noticed that now that I was looking more into it.  they are from the US government site here.
https://public.cyber.mil/stigs/gpo/ It uses https://github.com/jseerden/IntuneBackupAndRestore which has not been updated in 3 years, 
looks like that tool got updated in a diffrent branch.  Love your tool.  hope to use it more in the future.

https://github.com/mhu4711/IntuneBackupAndRestore/tree/mhu-mggraph-migration
